doc/source/toolchain.rst

```diff
@@ -0,0 +1,226 @@
+
+SciPy aims to be compatible with a number of releases of its dependent libraries and tools.
+Forcing the user base to other components for upgrade for every release would greatly
+diminish the value of SciPy.  However, maintaining backwards compatibility
+with very old tooling/libraries imposes limitations on which newer functionalities and capabilities
+can be incorporated.  SciPy takes a somewhat conservative approach, maintaining compatibility with
+several major releases of Python and NumPy on the major platforms. (That may in of itself impose
+further restrictions.  See the C Compilers section for an example.)
+
+
+- First and foremost, SciPy is a Python project hence it requires a Python environment.
+- The LAPACK and OpenBLAS numerical libraries need to be installed.
+- Compilers for C, C++, Cython and Fortran code are needed.
+- The Python environment needs the ``NumPy`` package to be installed.
+- Testing requires the ``pytest`` Python package.
+- Building the documentation requires the ``matplotlib``, Sphinx and ``numpydoc`` packages, as well as a LaTeX installation.
+
+The tooling used to build CPython has some implications for the tooling used in building SciPy.
+It also has implications for the documentation in docstrings.
+
+
+Building SciPy
+--------------
+
+Python Versions
+^^^^^^^^^^^^^^^
+
+SciPy is compatible with several versions of Python, and some
+specific decisions are still under consideration, especially
+with regard to future changes.
+Python 2.7 support was dropped for SciPy
+releases numbered 1.3 and above but is still available in Release 1.2.x,
+which is a long-term support release. [1]_, [2]_.
+
+================  =======================================================================
+ Date             Pythons supported
+================  =======================================================================
+ 2018              Py2.7, Py3.4+ (SciPy 1.2.x is the last release to support Python 2.7)
+ 2019              Py3.5+ (but Py2.7-specific code not removed)
+ 2020              Py3.5+ (removal of Py2.7-specific code permitted)
+================  =======================================================================
+
+NumPy
+^^^^^
+
+SciPy depends on NumPy but releases of SciPy are not tied to releases of NumPy.
+SciPy attempts to be compatible with at least the 4 previous releases of NumPy.
+In particular, SciPy can not rely on features of just the latest NumPy, but needs to be
+written using what is common in all of those 4 releases. [1]_, [6]_.
+
+========  ========================    ===========================
+ Python    Minimum NumPy version       Maximum NumPy version
+========  ========================    ===========================
+3.7         1.13.1                     >= 1.16.x
+3.6         1.12.1                     >= 1.16.x
+3.5         1.9.3                      >= 1.16.x
+2.7         1.8.2                      1.16.x
```

Comment:
we just updated these version numbers. Maybe put above this table that it's the status for SciPy 1.3.x. Then: 3.5/3.6 are numpy 1.13.3; py3.7 is numpy 1.14.5

Reply:
Done.